Campus-iOS/LoginComponent/Views/TokenPermissionsView.swift

```diff
@@ -0,0 +1,173 @@
+                    }
+                    //Identification
+                    if let currentState = viewModel.states[.identification] {
+                        check(state: currentState).padding()
+                    } else {
+                        Image(systemName: "questionmark.circle.fill").foregroundColor(.gray).padding()
+                    }
+                }
+            }
+            .font(.system(size: 20))
+            .padding()
+            
+            VStack (){
+                Button {
+                    self.showTUMOnline = true
+                } label: {
+                    HStack {
+                        Image(systemName: "globe")
+                        Text("Open TUMOnline")
+                    }
+                    .lineLimit(1).font(.body)
+                        .frame(width: 200, height: 48, alignment: .center)
+                }
+                .font(.title)
+                .foregroundColor(.white)
+                .background(Color(.tumBlue))
+                .cornerRadius(10)
+                
+                Button {
+                    Task {
+                        await viewModel.checkPermissionFor(types: [.grades, .lectures, .calendar, .identification, .tuitionFees])
+                        
+                        withAnimation() {
+                            doneButton = true
+                        }
+                    }
+                } label: {
+                    Text("Check Permissions")
+                        .lineLimit(1)
+                        .font(.body)
+                        .frame(width: 200, height: 48, alignment: .center)
+                        .foregroundColor(.white)
+                        .background(Color(.tumBlue))
+                        .cornerRadius(10)
+                        .buttonStyle(.plain)
+                }
+                
+                if doneButton {
+                    Button {
+                        if dismissWhenDone {
+                            // Dismiss when view is opened from Profile/Settings.
+                            dismiss()
+                        } else {
+                            // Used when shown via the login process sheet.
+                            self.viewModel.model.isLoginSheetPresented = false
+                        }
+                    } label: {
+                        Text("Done")
+                            .lineLimit(1)
+                            .font(.body)
+                            .frame(width: 200, height: 48, alignment: .center)
+                            .foregroundColor(.white)
+                            .background(.green)
+                            .cornerRadius(10)
+                            .buttonStyle(.plain)
+                    }
+                }
+            }
+        }
+        .sheet(isPresented: $showTUMOnline) {
+             SFSafariViewWrapper(url: URL(string: "https://www.campus.tum.de")!).edgesIgnoringSafeArea(.bottom)
```

Comment:
Same here, maybe direct to token view in campus online?

Reply:
Also, maybe move this url to a constant?

Reply:
✅